__tests__/db.test.js

```diff
@@ -56,4 +112,83 @@ describe("GET /api", () =>{
+describe("POST /api/articles/:article_id/comments", () =>{
+    test("Returns a 201 when posting data", ()=>{
+        return request(app).post("/api/articles/1/comments").send({username:"lurker", body:"Sorry -snip-"})
+            .then((res)=>{
+                expect(res.statusCode).toBe(201)
+        })
+    })
+    test("Returns only the comment when successfully posted", ()=>{
+        return request(app).post("/api/articles/1/comments").send({username:"lurker", body:"Sorry -snip-"})
+            .then((res)=>{
+                expect(res.statusCode).toBe(201)
+                expect(res.body.postedComment).toBe("Sorry -snip-")
+                expect(Object.keys(res.body)).toEqual(["postedComment"])
+        })
+    })
+    test("Returns 403 when given an invalid username", () =>{
+        return request(app)
+        .post('/api/articles/1/comments').send({username:'kasugaosaka', body: "OH MY GAH"})
+        .then((res)=>{
+                expect(res.statusCode).toBe(403)
+        })
+    })
+    test("Returns a 404 when posting to an invalid article", () =>{
+        return request(app)
+        .post('/api/articles/50/comments').send({username:'lurker', body: "Um, how do I post again? edit: nvm"})
+        .then((res)=>{
+                expect(res.statusCode).toBe(404)
+        })
+    })
+})
+
+describe.only('PATCH /api/articles/:article_id', () =>{
+    test('Returns 200 when given a inc_votes object with numerical data', ()=>{
+        return request(app)
+        .patch('/api/articles/4').send({inc_votes: 1})
+        .then((res)=>{
+                expect(res.statusCode).toBe(200)
+        })
+    })
+    test('Returns full article with the increased vote', ()=>{
+        return request(app)
+        .patch('/api/articles/1').send({inc_votes: 50})
+        .then((res)=>{
+                expect(res.body.votes).toBe(150)
+                expect(res.body.article_id).toBe(1)
+                expect(res.body.title).toBe('Living in the shadow of a great man')
+        })
+    })
+    test('accepts decreasing the vote', ()=>{
+        return request(app)
+        .patch('/api/articles/1').send({inc_votes: -50})
+        .then((res)=>{
+                expect(res.body.votes).toBe(100)
+                expect(res.body.article_id).toBe(1)
+        })
+    })
+    test('returns 400 when giving bad content', ()=>{
+        return request(app)
+        .patch('/api/articles/1').send({inc_votes: "Twenny"})
+        .then((res)=>{
+                expect(res.statusCode).toBe(400)
+        })
+    })
+    test('Returns 400 when given no content', ()=>{
+        return request(app)
+        .patch('/api/articles/1').send()
+        .then((res)=>{
+                expect(res.statusCode).toBe(400)
+        })
+    })
+    test('Returns 404 when given invalid article', ()=>{
+        return request(app)
+        .patch('/api/articles/999999').send({inc_votes: 1})
+        .then((res)=>{
+                expect(res.statusCode).toBe(404)
+        })
```

Comment:
What would happen if inc_votes was missing? (it would be fine! - a 200 and respond with the comment that has not been updated)